core/execution/engine_snapshot.go

```diff
@@ -104,12 +109,15 @@ func (e *Engine) restoreMarket(ctx context.Context, em *types.ExecMarket) (*Mark
 		return nil, err
 	}
 
+	mkt.nextMTM = nextMTM
 	e.markets[marketConfig.ID] = mkt
 	e.marketsCpy = append(e.marketsCpy, mkt)
 
 	if err := e.propagateInitialNetParams(ctx, mkt); err != nil {
 		return nil, err
 	}
+	// ensure this is set correctly
+	mkt.nextMTM = nextMTM
```

Comment:
is line 112 not sufficiently set correctly?

Reply:
I don't think so. I think line 112 is redundant, if anything. By calling `propagateInitialNetParams` the `nextMTM` time will get reset to be N seconds in the future. When restoring from a snapshot we have to make sure we'll MTM the exact same time as other nodes did, so the snapshot could have markets with a next MTM 2 seconds in the future, but the `propagateIntialNetParams` call would override that to be N seconds in the future (whatever the net param is)

Reply:
well, you the earlier assignment so now it all makes sense.